lib/embiggen/uri.rb

```diff
@@ -37,46 +29,22 @@ def shortened?
 
     private
 
-    def check_redirects(redirects)
-      return unless redirects.zero?
-
-      fail TooManyRedirects.new("#{uri} redirected too many times", uri)
-    end
-
-    def check_location(location)
-      return if location
+    def extract_redirects(request_options = {})
+      redirects = request_options.fetch(:redirects) { Configuration.redirects }
+      fail TooManyRedirects.new(
+        "#{uri} redirected too many times", uri) if redirects.zero?
```

Comment:
Correct me if I'm wrong, but at this point `uri` is not the original one anymore, so the message is not entirely correct.